test/overseer/heartbeat_test.clj

```diff
@@ -0,0 +1,57 @@
+(ns overseer.heartbeat-test
+  (:require [clojure.test :refer :all]
+            [clj-time.core :as clj-time]
+            [framed.std.time :as std.time]
+            [taoensso.timbre :as timbre]
+            [overseer.api :as api]
+            [overseer.core :as core]
+            [overseer.heartbeat :as heartbeat]
+            [overseer.test-utils :as test-utils]))
+
+(defn test-start-heartbeat [store]
+  (timbre/with-log-level :report
+    (let [config {:heartbeat {:monitor-shutdown false}}
+          old-heartbeat
+          (-> (clj-time/now)
+              (clj-time/minus (clj-time/days 30))
+              std.time/datetime->unix)
+          {job-id :job/id :as job} (test-utils/job {:job/heartbeat old-heartbeat})
+          _ (core/transact-graph store (api/simple-graph job))
+          current-job (atom (core/job-info store job-id))
+
+          heartbeat-fut (heartbeat/start-heartbeat config store current-job)]
+      (try
+        (Thread/sleep 500)
```

Comment:
I agree we need a better story here, but I'm not sure I understand the suggestion - the challenge is monitoring the effects of a thread that is difficult to instrument as far as I know, i.e. it doesn't accept a channel used to communicate anything, for instance, nor a user-defined job-handler like when testing executors.